varcode/common.py

```diff
@@ -59,3 +59,14 @@ def trim_shared_flanking_strings(ref, alt):
     ref, alt, prefix = trim_shared_prefix(ref, alt)
     ref, alt, suffix = trim_shared_suffix(ref, alt)
     return ref, alt, prefix, suffix
+
+
+def group_by(records, field_name):
```

Comment:
I'd imagine you're doing this instead of using itertools.group_by because that one returns generators instead? As an aside, this is a great place to use a collections.defaultdictionary.